### PR TITLE
csmith: init at 2.3.0, pull in Sys::CPU as dependency.

### PR DESCRIPTION
**Pull request by dtzWill**
###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"` (new package, there aren't any)
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/tools/misc/csmith/default.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    sha256 = "1mb5zgixsyf86slggs756k8a5ddmj980md3ic9sa1y75xl5cqizj";
+  };
+
+  nativeBuildInputs = [ m4 makeWrapper ];
+  buildInputs = [ libbsd perl SysCPU ];
+
+  postInstall = ''
+    substituteInPlace $out/bin/compiler_test.pl \
+      --replace '$CSMITH_HOME/runtime' $out/include/${name} \
+      --replace ' ''${CSMITH_HOME}/runtime' " $out/include/${name}" \
```
Comment (pSub) [inline pkgs/development/tools/misc/csmith/default.nix]: Maybe I am wrong but those single quotes seem wrong. I guess you mean something like  `--replace '${CSMITH_HOME}/runtime' " $out/include/${name}" \ `

File: pkgs/development/tools/misc/csmith/default.nix [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    mv $out/bin/compiler_test.in $out/share/csmith/
+  '';
+
+  enableParallelBuilding = true;
+
+  meta = with stdenv.lib; {
+    description = "A random generator of C programs";
+    homepage = "https://embed.cs.utah.edu/csmith";
+    # Officially, the license is this: https://github.com/csmith-project/csmith/blob/master/COPYING
+    license = licenses.bsd;
```
Comment (pSub) [inline pkgs/development/tools/misc/csmith/default.nix]: `licenses.bsd` is not a valid option. This looks like a 2-clause BSD license with is called `licenses.bsd2`. See https://github.com/NixOS/nixpkgs/blob/master/lib/licenses.nix#L78

**Review by pSub — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/misc/csmith/default.nix (hunk 1 above)
Comment (dtzWill) [inline pkgs/development/tools/misc/csmith/default.nix]: Basically, but if written that way (`--replace '${CSMITH_HOME}/runtime') Nix will try to substitute for a 'CSMITH_HOME' variable, here I'm basically escaping it so the script actually contains the dollar-sign and brace characters.

File: pkgs/development/tools/misc/csmith/default.nix (hunk 2 above)
Comment (dtzWill) [inline pkgs/development/tools/misc/csmith/default.nix]: I didn't think so, but I thought it'd break if it didn't exist :3.  Thanks, I'll fix that!

Comment (pSub) [inline pkgs/development/tools/misc/csmith/default.nix]: It broke. Look at the last travis build :-)

File: pkgs/development/tools/misc/csmith/default.nix (hunk 1 above)
Comment (pSub) [inline pkgs/development/tools/misc/csmith/default.nix]: Ah, I see. That makes sense.

File: pkgs/development/tools/misc/csmith/default.nix (hunk 2 above)
Comment (dtzWill) [inline pkgs/development/tools/misc/csmith/default.nix]: Interesting, well regardless I should've checked :).  But I'll see what it's doing to test things that I'm not! :+1:

**Comment by Mic92** [comment]
Thanks!

**Pull request merged**